### ISSUE TITLE
Loki 2.9 memberlist issue when deploying with loki-distributed chart

### ISSUE THREAD

**Issue by andy2809** [grafana/loki#10797]
**Describe the bug**
When specifying an IP address to bind using the flag -memberlist.bind-addr, the resulting config ends up with two entries of the specified IP. Then Loki 2.9 fails to start because it will attempt to bind the same address twice, the second time failing with:

> level=error ts=2023-10-04T07:05:33.696841705Z caller=loki.go:517 msg="module failed" module=memberlist-kv error="invalid service state: Failed, expected: Running, failure: failed to create transport: failed to start TCP listener on \"11.32.28.15\" port 7946: listen tcp 11.32.28.15:7946: bind: address already in use"

Seems to be the same problem as described in this SOLVED bug: https://github.com/grafana/loki/issues/2363

This worked with loki-distributed version <= 0.69.16 (aka Loki 2.8.x)

**To Reproduce**
1. configure ring with memberlist, add container flag -memberlist.bind-addr
2. deploy loki-distributed helm chart Version >= 0.70.x (aka Loki 2.9.x)

**Expected behavior**
Loki Components get started successfully. 

**Environment:**
 - Kubernetes Cluster v1.24
 - Deployment tool: loki-distributed helm-chart

**Screenshots, Promtail config, or terminal output**
If applicable, add any output to help explain your problem.

[ingester-loki-2.9.log](https://github.com/grafana/loki/files/12819691/ingester-loki-2.9.log)

**Comment by rhyswilliamsza** [comment]
I am experiencing this issue, too!

**Comment by cdm-anv** [comment]
No, update on this? Seems to be a common issue. Could you re-produce this problem?

**Comment by butschi84** [comment]
I use loki 2.9.1 and face the same problem. (loki distributed chart 0.75)
I put a "dirty hack" in the structuredConfig and that seems to be a workaround for the issue:

``` 
 structuredConfig:
    memberlist:
      bind_addr: []
```

**Comment by rhyswilliamsza** [comment]
Thanks @butschi84!

**Comment by gyoza** [comment]
> structuredConfig:
>     memberlist:
>       bind_addr: []

I love you
